packages/pages/src/common/src/parsers/templateParser.test.ts

```diff
@@ -68,10 +68,83 @@ describe("makeClientTemplate", () => {
       `import { Template, TemplateRenderProps } from "@yext/pages";`
     );
   });
+
+  it("does not return tsdoc comments", () => {
+    const sourceParser = createParser(
+      `/**
+      * mock tsdoc comments
+      */
+      const foo = 5;
+      export default foo`
+    );
+    const testParser = createParser(``);
+    const templateParser = new TemplateParser(
+      sourceParser
+    ).makeClientTemplateFromSfp(testParser);
+    expect(templateParser.fileContents).not.toContain(` mock tsdoc comments`);
+  });
+});
+
+describe("complex makeClientTemplate usages", () => {
+  it("correctly returns expected file contents", () => {
+    const sourceParser = new SourceFileParser(
+      "./tests/fixtures/sourceFileTemplates/static.tsx",
+      createTsMorphProject()
+    );
+    const expectedParser = new SourceFileParser(
+      "./tests/fixtures/sourceFileTemplates/clientStatic.tsx",
+      createTsMorphProject()
+    );
+    const testParser = createParser(``);
+    const templateParser = new TemplateParser(
+      sourceParser
+    ).makeClientTemplateFromSfp(testParser);
+    expectedStaticImports.forEach((expectedImport) => {
+      expect(templateParser.fileContents).toContain(expectedImport);
+    });
+    unnecessaryStaticStrings.forEach((unnecessaryString) => {
+      expect(templateParser.fileContents).not.toContain(unnecessaryString);
+    });
+    expect(templateParser.fileContents).toContain(expectedParser.getAllText());
+  });
+
+  it("correctly returns nothing for templates without default export", () => {
+    const sourceParser = new SourceFileParser(
+      "./tests/fixtures/sourceFileTemplates/robots.tsx",
+      createTsMorphProject()
+    );
+    const testParser = createParser(``);
+    const templateParser = new TemplateParser(
+      sourceParser
+    ).makeClientTemplateFromSfp(testParser);
+    expect(templateParser.fileContents).toEqual("");
+  });
 });
 
 function createParser(sourceCode: string) {
   const filepath = path.resolve(__dirname, "test.tsx");
   const { project } = createTestSourceFile(sourceCode, filepath);
   return new SourceFileParser(filepath, project);
 }
+
+const expectedStaticImports = [
+  `import { Template, TemplateRenderProps } from "@yext/pages";`,
+  `import Banner from `,
+  `import Card from `,
+  `import PageLayout from `,
+  `import { ExternalImage } `,
+];
+
+const unnecessaryStaticStrings = [
```

Comment:
I don't think having this is necessary. This will naturally be checked when you compare (input -> ts-morph) == client.